components/proxy/src/main/java/com/hotels/styx/infrastructure/FileBackedRegistry.java

```diff
@@ -92,15 +96,29 @@ public synchronized void reload(ReloadListener listener) {
     }
 
     private boolean updateResources() {
+        previousFileContent = fileContent;
+        fileContent = readFileAsString();
+
         Iterable<T> resources = reader.read(readFile());
```

Comment:
We are reading the file twice in succession (in lines 100 and 102). Please change this.